core/arch/arm/mm/mobj.c

```diff
@@ -385,12 +385,24 @@ static TEE_Result mobj_reg_shm_get_cattr(struct mobj *mobj __unused,
 	return TEE_SUCCESS;
 }
 
+static bool mobj_reg_shm_matches(struct mobj *mobj, enum buf_is_attr attr);
+
 static const struct mobj_ops mobj_reg_shm_ops __rodata_unpaged = {
 	.get_pa = mobj_reg_shm_get_pa,
 	.get_cattr = mobj_reg_shm_get_cattr,
+	.matches = mobj_reg_shm_matches,
 	.free = mobj_reg_shm_free,
 };
 
+static bool mobj_reg_shm_matches(struct mobj *mobj __maybe_unused,
+				   enum buf_is_attr attr)
+{
+	assert(mobj->ops == &mobj_reg_shm_ops);
+
+	return attr == CORE_MEM_NON_SEC || attr == CORE_MEM_NSEC_SHM ||
```

Comment:
`CORE_MEM_NSEC_SHM` objects do not belong to the registered SHM areas (as `CORE_MEM_REG_SHM` do) as they belong to the static pre-mapped shared memory. I wonder if this change may confuse shm identification when core is built without debug support.